storm-shaded-deps/pom.xml

```diff
@@ -297,6 +297,28 @@
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-pmd-plugin</artifactId>
             </plugin>
+            <plugin>
+                <groupId>org.codehaus.mojo</groupId>
+                <artifactId>build-helper-maven-plugin</artifactId>
+                <executions>
+                    <execution>
+                        <id>compile</id>
+                        <phase>none</phase>
+                        <goals>
+                            <goal>attach-artifact</goal>
+                        </goals>
+                        <configuration>
+                            <artifacts>
+                                <artifact>
+                                    <file>${basedir}/target/storm-shaded-deps-${project.version}.jar</file>
```

Comment:
Maybe use `<file>${project.build.directory}/${project.build.finalName}.jar</file>` ?

Reply:
This jar file has to be the one that was built (using the shade plugin). Can't be an arbitrary or unavailable jar AFAIK.

Reply:
Never mind - this works too. Those variables are defined in pom-4.0.0.xml.